### PR TITLE
Test runner for PRs

### PR DESCRIPTION
**Pull request by brett-groshong**
### Purpose of this PR

This change adds a yamato runner for PRs that start with the prefix `sg2`.

---
### Testing status

This PR should have the test runner results showing.

You can see the details of the execution by clicking on the links below in the "All checks" section.

image

### CODE REVIEW

**Comment by theo-pnv** [comment]
Hi @brett-groshong , we noticed there is a review request for our team (GFX SDETs). 
We don't think that our team should gate the shadergraph staging branches, so please feel free to move forward with the PR as you wish.

**Comment by brett-groshong** [comment]
@theo-pnv I have no idea how reviews were requested for this PR. Is there a naming convention that I should be aware of?

**Comment by patrickfournier** [comment]
@brett-groshong It happened to me too. GFX-SDET was automatically added to my PR because I mistakenly targeted the `master` branch.

File: .yamato/sg2-tests.yml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    UTR_VERSION: "current"
+    TEST_FILTER: .*
+    EDITOR_BRANCH: "sg2/2023-01-05-working-editor"
+  commands:
+    -  command: NetSh Advfirewall set allprofiles state off
+    -  command: curl -s https://artifactory.prd.it.unity3d.com/artifactory/unity-tools-local/utr-standalone/utr.bat --output utr.bat
+       retries: 2
+    -  command: choco install unity-downloader-cli -y -s https://artifactory.prd.it.unity3d.com/artifactory/api/nuget/unity-choco-local
+       retries: 2
+    -  command: unity-downloader-cli -u gtf/staging -c editor --fast
```
Comment (patrickfournier) [inline .yamato/sg2-tests.yml:25]: You are not getting"sg2/2023-01-05-working-editor".

File: .yamato/sg2-tests.yml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    flavor: m1.mac
+  variables:
+    UTR_VERSION: "current"
+    TEST_FILTER: .*
+    EDITOR_BRANCH: "sg2/2023-01-05-working-editor"
+  commands:
+    -  command: brew install lz4
+    -  command: brew tap --force-auto-update Unity-Technologies/homebrew-unity git@github.cds.internal.unity3d.com:unity/homebrew-unity.git
+    -  command: brew install Unity-Technologies/homebrew-unity/unity-downloader-cli
+    -  command: unity-downloader-cli -u gtf/staging -c editor --fast --wait
```
Comment (patrickfournier) [inline .yamato/sg2-tests.yml:47]: You are not getting"sg2/2023-01-05-working-editor".

Comment (brett-groshong) [reply .yamato/sg2-tests.yml]: Updated and moved into variable.

File: .yamato/sg2-tests.yml (hunk 1 above)
Comment (brett-groshong) [reply .yamato/sg2-tests.yml]: Updated and moved into variable.

File: com.unity.sg2/Tests/Generation/GenerationTests.cs [hunk 3]
```diff
@@ -86,32 +86,37 @@ private static Texture2D DrawToTex(Material material, int width = 4, int height
...
+        //         Assert.AreEqual(pixelColor, input.expectedColor);
+        //     }
+        //     catch(Exception e)
+        //     {
+        //         File.WriteAllBytes($"Assets/FailureImage{input.nodeToCompile}.jpg", rt.EncodeToJPG());
+        //         throw e;
+        //     }
+        // }
+
+        [Ignore("Disabled due to problems with Generation", Until="2023-01-7")]
```
Comment (patrickfournier) [inline com.unity.sg2/Tests/Generation/GenerationTests.cs:119]: Either you are using YYYY-DD-MM date format or all your dates are in the pass. I would think NUnit expects YYYY-MM-DD.

**Review by patrickfournier — CHANGES_REQUESTED** [review]
Date issue.

File: com.unity.sg2/Tests/Generation/GenerationTests.cs (hunk 3 above)
Comment (brett-groshong) [reply com.unity.sg2/Tests/Generation/GenerationTests.cs:119]: I'll try to update a date to the past and see it fail.

**Comment by theo-pnv** [comment]
> @theo-pnv I have no idea how reviews were requested for this PR. Is there a naming convention that I should be aware of?

Sorry I missed the comment about reviews.
It might be because of the codeowners. https://github.com/Unity-Technologies/Graphics/blob/sg2/main/.github/CODEOWNERS#L29
If you are using the sg2/main branch frequently and ever have the occasion to remove us from the codeowners on that branch, feel free to do so.

File: com.unity.sg2/Tests/Generation/GenerationTests.cs (hunk 3 above)
Comment (brett-groshong) [reply com.unity.sg2/Tests/Generation/GenerationTests.cs]: These are removed and passing now.

Comment (brett-groshong) [reply com.unity.sg2/Tests/Generation/GenerationTests.cs]: image

**Review by Nightmask3 — APPROVED** [review]
LGTM! 🟢

**Pull request merged**